pkg/apis/v1beta1/nodepool.go

```diff
@@ -98,20 +104,25 @@ type Budget struct {
 	// MaxUnavailable dictates how many NodeClaims owned by this NodePool
 	// can be terminating at once. It must be set.
 	// This only considers NodeClaims with the karpenter.sh/disruption taint.
-	// +kubebuilder:validation:XIntOrString
+	// We can't use an intstr.IntOrString since kubebuilder doesn't support pattern
+	// checking for int values for IntOrString values.
+	// Ref: https://github.com/kubernetes-sigs/controller-tools/blob/55efe4be40394a288216dab63156b0a64fb82929/pkg/crd/markers/validation.go#L379-L388
+	// +kubebuilder:validation:Type="string"
```

Comment:
Do you need this validation type now that you are explicitly defining this value as a string?